runtime/doc/lua.txt

```diff
@@ -1571,6 +1571,75 @@ schedule_wrap({cb})                                      *vim.schedule_wrap()*
       • |vim.schedule()|
       • |vim.in_fast_event()|
 
+system({spec}, {on_exit})                                       *vim.system()*
+    Run a system command
+
+    Examples: >lua
+
+      local on_exit = function(code, signal, stdout, stderr)
```

Comment:
- as @wsdjeg suggested, the pid seems worth including ere
- probably worth passing these args as a dict (named `event` or `args`?) instead of order-sensitive parameters? that seems like a good pattern whenever we anticipate callbacks that have more than ~3 args

Reply:
I'm not sure there's much value in the pid, given that when the callback fires the process is gone,  so you can't really do anything with the pid.

To distinguish between callbacks one can use closures.

Reply:
The goal here is to basically just consolidate all the uses of `uv.spawn` into a single API. I've not seen any users of `uv.spawn` need access to pid, and like @mfussenegger it isn't that useful and can easily be achieved with closures.

If there's a stronger case put forward in the future we can always add it.

Also this example is out of date it should now be:

```
local on_exit = function(obj)
```

Where `obj` has fields: `code`, `signal`, `stdout`, `stderr`.

Reply:
Since the args are no longer positional, this is a 2-way door.